src/material/schematics/ng-add/setup-project.ts

```diff
@@ -66,41 +68,106 @@ function addAnimationsModule(options: Schema) {
   return async (host: Tree, context: SchematicContext) => {
     const workspace = await getWorkspace(host);
     const project = getProjectFromWorkspace(workspace, options.project);
-    const appModulePath = getAppModulePath(host, getProjectMainFile(project));
-
-    if (options.animations === 'enabled') {
-      // In case the project explicitly uses the NoopAnimationsModule, we should print a warning
-      // message that makes the user aware of the fact that we won't automatically set up
-      // animations. If we would add the BrowserAnimationsModule while the NoopAnimationsModule
-      // is already configured, we would cause unexpected behavior and runtime exceptions.
-      if (hasNgModuleImport(host, appModulePath, noopAnimationsModuleName)) {
-        context.logger.error(
-          `Could not set up "${browserAnimationsModuleName}" ` +
-            `because "${noopAnimationsModuleName}" is already imported.`,
-        );
-        context.logger.info(`Please manually set up browser animations.`);
+
+    try {
+      addAnimationsModuleToNonStandaloneApp(host, project, context, options);
+    } catch (e) {
+      if (e instanceof SchematicsException && e.message.includes('Bootstrap call not found')) {
+        addAnimationsModuleToStandaloneApp(host, project, context, options);
       } else {
-        addModuleImportToRootModule(
-          host,
-          browserAnimationsModuleName,
-          '@angular/platform-browser/animations',
-          project,
-        );
+        throw e;
       }
-    } else if (
-      options.animations === 'disabled' &&
-      !hasNgModuleImport(host, appModulePath, browserAnimationsModuleName)
-    ) {
-      // Do not add the NoopAnimationsModule module if the project already explicitly uses
-      // the BrowserAnimationsModule.
+    }
+  };
+}
+
+/** Adds the animations module to an app that is bootstrap using the standalone component APIs. */
+function addAnimationsModuleToStandaloneApp(
+  host: Tree,
+  project: ProjectDefinition,
+  context: SchematicContext,
+  options: Schema,
+) {
+  const mainFile = getProjectMainFile(project);
+
+  if (options.animations === 'enabled') {
+    // In case the project explicitly uses the NoopAnimationsModule, we should print a warning
+    // message that makes the user aware of the fact that we won't automatically set up
+    // animations. If we would add the BrowserAnimationsModule while the NoopAnimationsModule
+    // is already configured, we would cause unexpected behavior and runtime exceptions.
+    if (importsProvidersFrom(host, mainFile, noopAnimationsModuleName)) {
+      context.logger.error(
+        `Could not set up "${browserAnimationsModuleName}" ` +
+          `because "${noopAnimationsModuleName}" is already imported.`,
```

Comment:
nit: can we include information on where the `NoopAnimationsModule` is imported?
```suggestion
          `because "${noopAnimationsModuleName}" is already imported into ` +
          `the "${...}" NgModule.`,
```